### PR TITLE
FAI-458 - Refactoring trusty infinispan storage: remove not necessary marshallers

### PR DESCRIPTION
**Pull request by r00ta**
https://issues.redhat.com/browse/FAI-458

All the objects we store in infinispan follow the new structure: 
1) first we have the properties we would like to index
2) the serialization of the entire object

In the marshaller we write the indexed properties and we serialize the object under the `rawObject` proto property. In the unmarshaller, we just read the `rawObject` string and we deserialize it. 

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Review by manstis — APPROVED** [review]
LGTM. Thank-you.

It might be _nice_ to add a `package-info.java` file [here](https://github.com/kiegroup/kogito-apps/tree/2f04453b7effee51cee3d4b212db6b214640f647/trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan) so we don't forget how/why it works this new way in the future; when we're adding new storage and new types and forget to index the fields we want to query elsewhere.

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java [hunk 1]
```diff
@@ -17,53 +17,30 @@
...
-import org.kie.kogito.trusty.storage.api.model.DecisionInput;
-import org.kie.kogito.trusty.storage.api.model.DecisionOutcome;
 import org.kie.kogito.trusty.storage.api.model.Execution;
-import org.kie.kogito.trusty.storage.api.model.ExecutionType;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class DecisionMarshaller extends AbstractModelMarshaller<Decision> {
 
+    private final ObjectMapper mapper;
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java]: What about add a `protected getMapper()` in [`AbstractMarshaller`](https://github.com/kiegroup/kogito-apps/blob/2f04453b7effee51cee3d4b212db6b214640f647/persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/protostream/AbstractMarshaller.java#L29) and avoid to duplicate the field here?

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/LIMEExplainabilityResultMarshaller.java [hunk 2]
```diff
@@ -16,35 +16,29 @@
...
 import org.kie.kogito.trusty.storage.api.model.BaseExplainabilityResult;
-import org.kie.kogito.trusty.storage.api.model.ExplainabilityStatus;
 import org.kie.kogito.trusty.storage.api.model.LIMEExplainabilityResult;
-import org.kie.kogito.trusty.storage.api.model.SaliencyModel;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class LIMEExplainabilityResultMarshaller extends AbstractModelMarshaller<LIMEExplainabilityResult> {
 
+    private final ObjectMapper mapper;
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/LIMEExplainabilityResultMarshaller.java]: Same here

File: trusty/trusty-storage/trusty-storage-infinispan/src/test/java/org/kie/kogito/trusty/storage/infinispan/MarshallerTestTemplate.java [hunk 3]
```diff
@@ -16,99 +16,41 @@
...
-
-        streamAllNonStaticFields(modelClass).forEach(field -> {
-            String serializedFieldName = field.isAnnotationPresent(JsonProperty.class)
-                    ? field.getAnnotation(JsonProperty.class).value()
-                    : field.getName();
 
-            long matches = testFieldNameList.stream().filter(n -> n.equals(serializedFieldName)).count();
-            assertEquals(1, matches, () -> String.format("Field \"%s\" of %s model is not handled properly in the corresponding test", serializedFieldName, modelClass.getSimpleName()));
-        });
+abstract class MarshallerTestTemplate {
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-infinispan/src/test/java/org/kie/kogito/trusty/storage/infinispan/MarshallerTestTemplate.java]: `public`?

**Comment by r00ta** [comment]
> LGTM. Thank-you.
> 
> It might be _nice_ to add a `package-info.java` file [here](https://github.com/kiegroup/kogito-apps/tree/2f04453b7effee51cee3d4b212db6b214640f647/trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan) so we don't forget how/why it works this new way in the future; when we're adding new storage and new types and forget to index the fields we want to query elsewhere.

Hi @manstis , I fully agree with you. What about a README.md in the trusty-storage-infinispan folder instead of the `package-info.java`?

**Comment by manstis** [comment]
> What about a README.md in the trusty-storage-infinispan folder instead of the `package-info.java`?

@r00ta Sounds good to me.. something is better than nothing :+1:

File: trusty/trusty-storage/trusty-storage-infinispan/src/test/java/org/kie/kogito/trusty/storage/infinispan/MarshallerTestTemplate.java [hunk 4]
```diff
@@ -16,99 +16,41 @@
...
+    protected MessageMarshaller.ProtoStreamReader reader;
+    protected Map<String, String> stringStorage;
+    protected Map<String, Long> longStorage;
+
+    @BeforeEach
+    protected void setup() throws IOException {
+        this.stringStorage = new HashMap<>();
+        this.longStorage = new HashMap<>();
+        this.writer = mock(MessageMarshaller.ProtoStreamWriter.class);
+        this.reader = mock(MessageMarshaller.ProtoStreamReader.class);
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-infinispan/src/test/java/org/kie/kogito/trusty/storage/infinispan/MarshallerTestTemplate.java:41]: Are you sure it is really necessary to mock these? I would prefer to use the real writer/reader

**Review by danielezonca — COMMENTED** [review]
Thanks for the PR, it is way more clear now :)

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 1 above)
Comment (manstis) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java]: Good spot @danielezonca. 

The field in the super class is already [protected](https://github.com/kiegroup/kogito-apps/blob/2f04453b7effee51cee3d4b212db6b214640f647/persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/protostream/AbstractMarshaller.java#L29) so we should be able to simply delete the private field here.

File: trusty/trusty-storage/trusty-storage-infinispan/src/test/java/org/kie/kogito/trusty/storage/infinispan/MarshallerTestTemplate.java (hunk 4 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/test/java/org/kie/kogito/trusty/storage/infinispan/MarshallerTestTemplate.java:41]: As per chat the initialization of the `ProtoStreamWriter` and `ProtoStreamReader` seems not to be trivial so I preferred to mock it for simplicity. Resolving the conv

**Comment by r00ta** [comment]
Jenkins retest this please

**Review by danielezonca — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**